File: tournaments/fractionReducing/fractionReducing.js
function fractionReducing(fraction) {
    const getGcd = function(a, b) {
        if (!a) return b;
        return getGcd(b%a, a);
    };

    const gcd = getGcd(fraction[0], fraction[1]);

    return [fraction[0]/gcd, fraction[1]/gcd];
}

File: tournaments/fractionReducing/fractionReducing.pl
#!/usr/bin/perl
use strict;
use warnings FATAL => 'all';

sub fractionReducing {
    my ($fraction) = @_;

    my $gcd = getGcd($$fraction[0], $$fraction[1]);

    return [$$fraction[0]/$gcd, $$fraction[1]/$gcd];
}

sub getGcd {
    my ($x, $y) = @_;

    unless ($x) {
        return $y;
    }

    return getGcd($y % $x, $x);
};
